src/transformers/generation/utils.py

```diff
@@ -2187,12 +2196,7 @@ def _contrastive_search(
 
             # stop when each sentence is finished
             unfinished_sequences = unfinished_sequences & ~stopping_criteria(input_ids, scores)
-
-            if unfinished_sequences.max() == 0:
-                this_peer_finished = True
-
-            if this_peer_finished and not synced_gpus:
-                break
+            this_peer_finished = unfinished_sequences.max() == 0
```

Comment:
The previous version is also a data-dependent control flow, so this change is for `torch.compile` readiness :)